### ISSUE TITLE
Can't use type parameter

### ISSUE THREAD

**Issue by nananapo** [veryl-lang/veryl#958]
```sv
module Top #(
    param param_type : type = logic,
) {
    type type_type = logic<32>;

    struct struct_type {
        a: logic,
    }

    inst m: ModuleA #(
      T1: param_type, // ok
      T2: type_type, // ng
      T3: struct_type, // ng
      T4: logic // ng
    );
}

module ModuleA #(
    param T1: type = logic,
    param T2: type = logic,
    param T3: type = logic,
    param T4: type = logic,
) {}

```

**Comment by dalance** [comment]
T2 and T3 will be fixed by #962.
T4 requires more consideration in #960.

**Comment by nananapo** [comment]
```$size(data_type)``` also results in same error.
